src/main/resources/org/apidesign/polyfill/websocket/websocket-polyfill.js

```diff
@@ -1,24 +1,221 @@
 (function (jvm) {
 
-    globalThis.WebSocket = function(config) {
-        debugger;
-        return {};
+    var EventBase = {
+        bubbles: false,
+        cancelable: false
     };
 
-    globalThis.WebSocketServer = function(config) {
+    var WebSocket = (function () {
+
+        function WebSocket(url, protocols) {
+            this.state_ = WebSocket.CONNECTING;
+
+            this.url_ = url;
+
+            if (typeof protocols === "string") {
+                protocols = [protocols];
+            }
+            this.protocols_ = protocols;
+
+            this.connection_ = jvm(
+                'new-web-socket-connection',
+                url,
+                protocols,
+                this._handle_open.bind(this),
+                this._handle_close.bind(this),
+                this._handle_error.bind(this),
+                this._handle_message.bind(this)
+            );
+        }
+
+        //
+        // Constants
+        //
+
+        WebSocket.CONNECTING = 0;
+        WebSocket.OPEN = 1;
+        WebSocket.CLOSING = 2;
+        WebSocket.CLOSED = 3;
+
+        //
+        // Properties
+        //
+
+        Object.defineProperty(WebSocket.prototype, "url", {
```

Comment:
Using `defineProperty`! Why? To control `enumerable`, etc.?

Reply:
Just to make the field read-only